usaspending_api/search/tests/integration/spending_by_category/test_spending_by_recipient.py

```diff
@@ -0,0 +1,203 @@
+import logging
+import uuid
+
+from model_mommy import mommy
+from elasticsearch_dsl import A
+
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.search.tests.integration.spending_by_category.utilities import setup_elasticsearch_test
+
+
+def _make_fpds_transaction(id, award_id, obligation, action_date, recipient_duns, recipient_name):
+    # Transaction Normalized
+    mommy.make(
+        "awards.TransactionNormalized",
+        id=id,
+        is_fpds=True,
+        award_id=award_id,
+        federal_action_obligation=obligation,
+        action_date=action_date,
+    )
+
+    # Transaction FPDS
+    mommy.make(
+        "awards.TransactionFPDS",
+        transaction_id=id,
+        awardee_or_recipient_uniqu=recipient_duns,
+        awardee_or_recipient_legal=recipient_name,
+    )
+
+
+def test_top_1_fails_with_es_transactions_routed_dangerously(client, monkeypatch, elasticsearch_transaction_index, db):
+    """
+    This confirms vulnerability of high-cardinality aggregations documented in DEV-4685, that leads to inaccurate
+    summing and ordering of sums when taking less buckets than the term cardinality.
+
+    This is shown by manually applying a routing key (using a key value stuck in ``awards.piid`` field here as the
+    routing key value) on index do that documents are distributed as below
+
+    NOTE: This requires an ES cluster with at least 3 shards for the transaction index. Ours should be defaulted to 5.
+
+    Recipient shard0   shard1   shard2   shard3   shard4
+    Biz 1      $2.00
+    Biz 1                       $ 7.00
+    Biz 1                       $ 3.00
+    Biz 1              $ 2.00
+    Biz 1              $ 3.00
+    Biz 1              $ 5.00
+    Biz 2              $ 6.00
+    Biz 2              $ 3.00
+    Biz 2                       $ 2.00
+    Biz 2                       $ 3.00
+    Biz 2                       $ 4.00
+    Biz 2     $13.00
+
+    **IF THIS TEST FAILS**
+        - Did our cluster structure change to not be 5 shards per the transaction index?
+        - Did the transaction<->award DB linkage change?
+        - Did we change ES version or config?
+            - Investigate if Elasticsearch has changed the way they do routing or hash routing key values
+    """
+
+    # Setup data for this test
+    recipient1 = uuid.uuid4()
+    recipient2 = uuid.uuid4()
+
+    # Recipient Lookup
+    mommy.make("recipient.RecipientLookup", id=1, recipient_hash=recipient1, legal_business_name="Biz 1", duns="111")
+    mommy.make("recipient.RecipientLookup", id=2, recipient_hash=recipient2, legal_business_name="Biz 2", duns="222")
```

Comment:
@tony-sappe This test relies on injecting specific recipient_hash values to the ES docs.
Let me know if data setup in this way won't accurately populate the `recipient_hash` value in the `universal_transaction_matview` that is then sent on to the transactions index in ES.

Reply:
If the tests work now, then injecting `recipient_hash` values is fine.